desktop/app/src/main/kotlin/com/abdownloadmanager/desktop/pages/home/HomeComponent.kt

```diff
@@ -900,4 +901,51 @@ class HomeComponent(
     companion object {
         val CATEGORIES_SIZE_RANGE = 0.dp..500.dp
     }
-}
\ No newline at end of file
+
+    private val fileWatcherScope = CoroutineScope(SupervisorJob() + Dispatchers.IO)
+    private val fileWatcher = FileSystems.getDefault().newWatchService()
+
+    init {
+        fileWatcherScope.launch {
+            while (true) {
+                val key = fileWatcher.take()
+                for (event in key.pollEvents()) {
+                    val kind = event.kind()
+                    if (kind == StandardWatchEventKinds.OVERFLOW) {
+                        continue
+                    }
+                    val ev = event as WatchEvent<Path>
+                    val fileName = ev.context()
+                    val filePath = (key.watchable() as Path).resolve(fileName)
+                    handleFileEvent(filePath)
+                }
+                key.reset()
+            }
+        }
+
+        scope.launch {
+            val downloadItems = downloadSystem.getDownloadList()
+            val downloadDirectories = downloadItems.map { it.folder }.distinct()
+            for (directory in downloadDirectories) {
+                val path = Paths.get(directory)
+                path.register(
+                    fileWatcher,
+                    StandardWatchEventKinds.ENTRY_DELETE,
+                    StandardWatchEventKinds.ENTRY_MODIFY,
+                    StandardWatchEventKinds.ENTRY_MOVE
+                )
+            }
+        }
+    }
+
+    private suspend fun handleFileEvent(filePath: Path) {
+        val downloadItems = downloadSystem.getDownloadList()
+        val affectedItems = downloadItems.filter {
+            val downloadFile = downloadSystem.getDownloadFile(it)
+            downloadFile.toPath() == filePath
+        }
+        for (item in affectedItems) {
+            downloadSystem.removeDownload(item.id, false)
+        }
+    }
+}
```

Comment:
Some notes here.

### `HomeComponent` is not a  good place to bootstrap long lived features
because it is has a lifecycle and may be closed / opened multiple times, which is causes to produce a lot of memory leaks!

### Organize the feature you want to add in a package/class.

You can create a file named `DownloadedFileWatcher` in some package
for example in `project:shared:app-utils` > `<appPacakge>/downloadfilewatcher/DownloadedFileWatcher.kt`
and do what you want to do in there.
instead of directly writing code in the `DownloadManager` and `HomeComponent` class (which both of them meant to be minimal as possible specially the first one!) to getting download list! 
- You can inject `DownloadMonitor` to your class to retrieve download list and also their folders
- You can inject ListOfJobsEvents in case you want to be notified about download list change (for example something started/ removed/ changed etc..)
- You can use `DownloadSystem` which is used for apps specific tasks that also related to the download manager itself. (I'm using these classes to making an abstraction between the app layer and the `Downloader` layer.)

- And finally implement your logic in your own class.
and boot the class in a place that have a lifecycle tied to the app(if its required to be survived until the user actually kills the process)


for example you can see the implementation of `Category Manager` and `ProxyManager` which they are all placed in `app-utils` module with their own packages!.